### PR TITLE
EZP-31079: Provided validation for email uniqueness and login pattern

### PR DESCRIPTION
**Pull request by Nattfarinn**
| Question      | Answer
| ------------- | ---
| Tickets       |  [EZP-31079](https://jira.ez.no/browse/EZP-31079) 
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

#### Checklist:
- [x] Implement tests
- [x] Coding standards (`$ composer fix-cs`)

### CODE REVIEW

File: src/bundle/Command/AuditUserDatabaseCommand.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+use Doctrine\DBAL\FetchMode;
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\Core\FieldType\User\Type;
+use eZ\Publish\Core\FieldType\User\UserStorage\Gateway\DoctrineStorage;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class AuditUserDatabaseCommand extends Command
```
Comment (adamwojs) [inline src/bundle/Command/AuditUserDatabaseCommand.php]:
```suggestion
final class AuditUserDatabaseCommand extends Command
```

File: src/bundle/Command/AuditUserDatabaseCommand.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+        $requireUniqueEmail = false;
+        foreach ($userFieldDefinitions as $userFieldDefinition) {
+            if ($userFieldDefinition->fieldSettings[Type::REQUIRE_UNIQUE_EMAIL]) {
+                $requireUniqueEmail = true;
+                break;
+            }
+        }
+
+        if (true || $requireUniqueEmail) {
```
Comment (adamwojs) [inline src/bundle/Command/AuditUserDatabaseCommand.php]: `true || $requireUniqueEmail` is always `true`

File: src/bundle/Command/AuditUserDatabaseCommand.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    ): int {
+        $userFieldDefinitions = $this->getUserFieldDefinitions();
+
+        $requireUniqueEmail = false;
+        foreach ($userFieldDefinitions as $userFieldDefinition) {
+            if ($userFieldDefinition->fieldSettings[Type::REQUIRE_UNIQUE_EMAIL]) {
+                $requireUniqueEmail = true;
+                break;
+            }
+        }
```
Comment (adamwojs) [inline src/bundle/Command/AuditUserDatabaseCommand.php]: Could be moved to private method eg  `isUniqeEmailRequired`

File: src/bundle/Command/AuditUserDatabaseCommand.php [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+class AuditUserDatabaseCommand extends Command
+{
+    /** @var \eZ\Publish\API\Repository\ContentTypeService */
+    private $contentTypeService;
+
+    /** @var \eZ\Publish\API\Repository\UserService */
+    private $userService;
+
+    /** @var \Doctrine\DBAL\Connection */
+    private $connection;
```
Comment (adamwojs) [inline src/bundle/Command/AuditUserDatabaseCommand.php]: Does it work with multi-repository setup ?

Comment (Nattfarinn) [reply src/bundle/Command/AuditUserDatabaseCommand.php]: Will it resolve multirepository setup issues: https://github.com/ezsystems/ezplatform-user/pull/62/files#diff-f228b61266937fbf4cc606c4ba771e95R29 ?

File: src/bundle/Command/AuditUserDatabaseCommand.php [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    {
+        $userFieldDefinitions = [];
+
+        $contentTypeGroups = $this->contentTypeService->loadContentTypeGroups();
+        foreach ($contentTypeGroups as $contentTypeGroup) {
+            $contentTypes = $this->contentTypeService->loadContentTypes($contentTypeGroup);
+
+            foreach ($contentTypes as $contentType) {
+                $fieldDefinitions = $contentType->getFieldDefinitionsOfType('ezuser');
+                if ($fieldDefinitions->count()) {
```
Comment (adamwojs) [inline src/bundle/Command/AuditUserDatabaseCommand.php]:
```suggestion
                if (!$fieldDefinitions->isEmpty()) {
```

File: src/bundle/Command/AuditUserDatabaseCommand.php (hunk 4 above)
Comment (alongosz) [reply src/bundle/Command/AuditUserDatabaseCommand.php]: > Will it resolve multirepository setup issues: https://github.com/ezsystems/ezplatform-user/pull/62/files#diff-f228b61266937fbf4cc606c4ba771e95R29 ?

Yes, `@ezpublish.persistence.connection` is built based on current Repository configured for SA

**Pull request merged**